setup.py

```diff
@@ -2447,8 +2447,13 @@ def split_var(name, sep):
         # features like DSO engines or external OSSL providers don't work.
         # Only tested on GCC and clang on X86_64.
         if os.environ.get("PY_UNSUPPORTED_OPENSSL_BUILD") == "static":
+            import subprocess, shutil
+            pkgconfig = os.getenv("PKG_CONFIG", shutil.which("pkg-config"))
+            libs_out = subprocess.check_output(
+                    [pkgconfig, "--static", "--libs-only-l", "openssl"]).decode()
```

Comment:
You are not taking ``--with-openssl=`` configure option into account. It will only look up ``openssl.pc`` in standard pkg-config paths.

Reply:
Where is that logic in the configure script? I can try to adapt it

Reply:
It's in an external autoconf-archive script called ``ax_check_openssl.m4``, https://www.gnu.org/software/autoconf-archive/ax_check_openssl.html . You cannot modify it.